### PR TITLE
implement credential joining for monitoring and deployment

### LINKED ISSUE #421: Implement credentials "Joining" feature for service deployment
- [ ] Credentials joining validation for service deployment must be implemented as described here - https://eclipse-xpanse.github.io/xpanse-website/docs/credentials-concept#service-deployment
- [ ] Additionally optional credential variables must be ignored during the "validation".
- [ ] unit tests

Note - based on the issue #399 , we will receive credential Type from request or we will receive receive the variables from request. To be decided. Rest of the requirements remain same.

### LINKED ISSUE #422: Implement credentials "joining" feature for monitoring
- [ ] Credentials to be used for making monitoring related API calls must be "collected" and "validated" as described here - 
https://eclipse-xpanse.github.io/xpanse-website/docs/credentials-concept#service-monitoring
- [ ] Additionally optional credential variables must be ignored during the "validation".
- [ ] Unittests

### PR DESCRIPTION
**Pull request by swaroopar**
fixes #421 
fixes #422

### CODE REVIEW

File: modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCenter.java [hunk 1]
```diff
@@ -314,7 +303,7 @@ public void createCredential(AbstractCredentialInfo abstractCredentialInfo) {
      * @param csp The cloud service provider.
      * @return Returns credentialInfo.
      */
-    private AbstractCredentialInfo findCredentialInfoFromEnv(Csp csp) {
+    private AbstractCredentialInfo getCompleteCredentialDefinitionFromEnv(Csp csp) {
         List<AbstractCredentialInfo> credentialAbilities =
                 getCredentialCapabilitiesByCsp(csp, null);
```
[the remark this answers is not in the document]
Comment (swaroopar) [reply modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCenter.java:308]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (swaroopar) [reply modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCenter.java:308]: @sonatype-lift ignore

**Review by baixinsui — APPROVED** [review]
LGTM.

**Review by iskey — APPROVED** [review]
lgtm

**Pull request merged**